x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/IndexResolver.java

```diff
@@ -84,58 +84,88 @@ public IndexResolver(Client client) {
     }
 
     /**
-     * Resolves a pattern to one (potentially compound meaning that spawns multiple indices) mapping.
+     * Like {@link #resolveAsMergedMapping(String, Set, QueryBuilder, boolean, boolean, boolean, ActionListener)},
+     * but uses a pre-determined minimum transport version.
      */
-    public void resolveAsMergedMapping(
-        String indexWildcard,
+    public void resolveAsMergedMappingForVersion(
+        String indexPattern,
         Set<String> fieldNames,
         QueryBuilder requestFilter,
         boolean includeAllDimensions,
         boolean useAggregateMetricDoubleWhenNotSupported,
         boolean useDenseVectorWhenNotSupported,
```

Comment:
Since you are changing the signature, those 3 booleans are always false when supplied to this method. The signature could be simplified.

Reply:
I removed this method again to simplify.

I considered whether the `use...WhenNotSupported` booleans can be removed, but they're still required in rolling upgrades when one of the nodes is 9.2.0. This is the only version that's already supporting `aggregate_metric_double` and `dense_vector`, but doesn't provide its transport version in field caps responses.